enum.c

```diff
@@ -871,49 +875,141 @@ ary_inject_op(VALUE ary, VALUE init, VALUE op)
+ *    # Sum of squares of elements added to 2.
+ *    (1..4).inject(2) {|sum, n| sum + n*n } # => 32
+ *
+ *  <b>Case 1: <tt>inject(symbol) -> obj</tt></b>
+ *
+ *  Given the single argument +symbol+ (which names a binary method) and no block:
+ *
+ *  - Combines elements of +self+ using the binary method:
+ *
+ *    - The first two operands are <tt>self[0]</tt> and <tt>self[1]</tt>.
+ *    - The next two operands are that result and <tt>self[2]</tt>.
+ *    - The next two operands are that result and <tt>self[3]</tt>.
+ *    - And so on.
+ *
+ *  - Returns the last result.
+ *
+ *  The returned object may not be of the same class as the first element.
+ *
+ *  Examples:
+ *
+ *    # Integer
+ *    (1..4).inject(:+)                # => 10
+ *    # Integer
+ *    [1, 2, 3.0, 4].inject(:+)        # => 10.0
+ *    #Float
+ *    [1.0, 2, 3, 4].inject(:+)        # => 10.0
+ *    # String
+ *    ('a'..'d').inject(:+)            # => "abcd"
+ *    # Complex
+ *    [Complex(1, 2), 3, 4].inject(:+) # => (8+2i)
+ *
+ *  Trivially, and regardless of the given method:
+ *
+ *  - <tt>self.inject</tt> for a 1-element +self+ returns that element.
+ *  - <tt>self.inject</tt> for an empty +self+ returns +nil+.
+ *
+ *  <b>Case 2: <tt>inject(initial_operand, symbol) -> obj</tt></b>
+ *
+ *  Given the two arguments +initial_operand+ and +symbol+ (which names a binary method)
+ *  and no block:
+ *
+ *  - Combines +initial+operand+ and elements of +self+ using the binary method:
+ *
+ *    - The first two operands +initial_operand+ and <tt>self[0]</tt> .
+ *    - The next two operands are that result and <tt>self[1]</tt>.
+ *    - The next two operands are that result and <tt>self[2]</tt>.
+ *    - And so on.
+ *
+ *  - Returns the last result.
+ *
+ *  The returned object may not be of the same class as the +initial_operand+.
+ *
+ *  Examples:
+ *
+ *    # Integer
+ *    (1..4).inject(2, :+)             # => 12
+ *    # Integer
+ *    [1, 2, 3.0, 4].inject(2, :+)     # => 12.0
+ *    #Float
+ *    (1..4).inject(2.0, :+)           # => 12.0
+ *    # String
+ *    ('a'..'d').inject('z', :+)       # => "zabcd"
+ *    # Complex
+ *    (1..4).inject(Complex(2, 3), :+) # => (12+3i)
+ *
+ *  Trivially, and regardless of the given method:
+ *
+ *  - <tt>self.inject</tt> for an empty +self+ returns +initial_operand+.
+ *
+ *  <b>Case 3: <tt>inject {|memo, element| ... } -> obj</tt></b>
+ *
+ *  Given no argument and a block:
+ *
+ *  - Calls the block repeatedly with two parameters:
+ *
+ *    - The parameters for the first call are <tt>self[0]</tt> and <tt>self[1]</tt>.
+ *    - The parameters for the next call are the block's return and <tt>self[2]</tt>.
+ *    - The parameters for the next call are the block's return and <tt>self[3]</tt>.
+ *    - And so on.
+ *
+ *  - Returns the last value returned by the block.
+ *
+ *  Examples:
+ *
+ *    # Return sum of squares.
+ *    (1..4).inject {|memo, n| memo + n*n } # => 30
```

Comment:
This example is misleading, it only returns the sum of squares if the square of the first element is the same as the element itself (for 0 and 1).  It does work for `(1..4)`, but not for `(2..4)`, for example.